src/Akka.Streams.Kafka.Tests/Integration/PlainSourceIntegrationTests.cs

```diff
@@ -151,7 +162,62 @@ private TestSubscriber.Probe<string> CreateProbe(ConsumerSettings<Null, string>
                 .WithGroupId(group1);
 
             var probe = CreateProbe(config, topic1, Subscriptions.Assignment(new TopicPartition(topic1, 0)));
-            probe.Request(1).ExpectError().Should().BeOfType<Exception>();
+            probe.Request(1).ExpectError().Should().BeOfType<KafkaException>();
+        }
+
+        [Fact]
+        public async Task PlainSource_should_stop_on_deserialization_errors()
+        {
+            int elementsCount = 10;
+            var topic1 = CreateTopic(1);
+            var group1 = CreateGroup(1);
+
+            await Produce(topic1, Enumerable.Range(1, elementsCount), ProducerSettings);
+
+            var settings = ConsumerSettings<Null, int>.Create(Sys, null, new IntDeserializer())
+                .WithBootstrapServers(KafkaUrl)
+                .WithProperty("auto.offset.reset", "earliest")
+                .WithGroupId(group1);
+
+            var probe = Dsl.Consumer
+                .PlainSource(settings, Subscriptions.Assignment(new TopicPartition(topic1, 0)))
+                .WithAttributes(ActorAttributes.CreateSupervisionStrategy(Deciders.StoppingDecider))
+                .Select(c => c.Value)
+                .RunWith(this.SinkProbe<int>(), _materializer);
+
+            var error = probe.Request(elementsCount).ExpectEvent(TimeSpan.FromSeconds(10));
+            error.Should().BeOfType<TestSubscriber.OnError>();
+            ((TestSubscriber.OnError)error).Cause.Should().BeOfType<SerializationException>();
```

Comment:
Streams TestKit has limited support for timeouts, so I can use only `ExpectEvent` and `ExpectNext`